kubernetes/schema_endpointslice.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func schemaEndpointSliceSubsetEndpoints() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:        schema.TypeList,
+				Description: "Name of the referent. More info: http://kubernetes.io/docs/user-guide/identifiers#names",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"conditions": {
+				Type:        schema.TypeList,
+				Description: "A list of references to secrets in the same namespace to use for pulling any images in pods that reference this Service Account. More info: http://kubernetes.io/docs/user-guide/secrets#manually-specifying-an-imagepullsecret",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ready": {
+							Type:        schema.TypeBool,
+							Description: "Specification of the desired behavior of the job",
+							Optional:    true,
+						},
+						"serving": {
+							Type:        schema.TypeBool,
+							Description: "Specification of the desired behavior of the job",
+							Optional:    true,
+						},
+						"terminating": {
+							Type:        schema.TypeBool,
+							Description: "Specification of the desired behavior of the job",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"host_name": {
```

Comment:
This can just be `hostname` the corresponding Kubernetes API field isn't camelCased.